renovate.json

```diff
@@ -0,0 +1,12 @@
+{
+  "extends": [
+    "config:base",
+    "schedule:weekends"
+  ],
+  "packageRules": [
+    {
+      "depTypeList": ["devDependencies"],
+      "extends": [":automergeMinor"]
```

Comment:
Even for a bot, I'd rather that we follow our [merging process](https://github.com/canonical-webteam/practices/blob/master/workflow/github.md#merging-process): "Your pull request must be approved by at least one person"

We wouldn't want to have anything broken even for minor updates.

Reply:
My thinking was that for `devDependencies` specifically, it would be fine to auto-merge them, for minor updates, because these are not dependencies that are ever run in our live applications - so they'll never introduce bugs to live applications, they'll only (possibly) introduce bugs to our development process, which I thought is acceptable.

But if you disagree, this feels like something that should be discussed with the team then. So I'll have to bring it up in dev catch-up.

Reply:
Leaving this as-is could still offer a convenience factor. It would mean:

- Renovate is told to automerge but cannot due to reviewer restrictions
- A reviewer can bulk review many PRs at a time and approve all that look fine
- Renovate will then automerge them one at a time, rebasing all remaining ones for re-testing if merging of the previous one resulted in merge conflicts

Without that, the reviewer could instead face the following scenario:
- Approve the first PR, merge it
- Wait for others to be rebased by the bot to resolve conflicts
- Wait for test to pass
- Repeat for each PR

Reply:
@rarkins I don't suppose there's any way to ask Renovate to group updates together in one PR? Because, assuming that normally these updates go through without a hitch, this might offer a smoother process.

Reply:
@nottrobin Renovate can group together any updates you wish if you configure that "statically" via Renovate config. For example you could always group together all non-major devDependency packages into a PR called "Update devDependency packages".

The natural downside of this is that "one bad apple spoils the bunch", so any time one update is bad, the PR is unmergeable and you'll to trace it down before you can move any of the others forward.

With the "config:base" preset included you would already see examples of that if you have any dependencies from known monorepos like react, jest, babel, etc - Renovate will group all those related ones together into related PRs.

Reply:
Okay, I'll try out the default settings on that to start with.

Reply:
Sorry I missread the comment you added about `devDependencies`. I don't have any problem with this.
Let's try it on this project and see how it goes :smile: 

thanks @rarkins  for the quick answers on the pull request!